datafusion/src/physical_plan/expressions/sum.rs

```diff
@@ -54,8 +56,15 @@ pub fn sum_return_type(arg_type: &DataType) -> Result<DataType> {
         DataType::UInt8 | DataType::UInt16 | DataType::UInt32 | DataType::UInt64 => {
             Ok(DataType::UInt64)
         }
-        DataType::Float32 => Ok(DataType::Float32),
-        DataType::Float64 => Ok(DataType::Float64),
+        // In the https://www.postgresql.org/docs/8.2/functions-aggregate.html doc,
+        // the result type of floating-point is FLOAT64 with the double precision.
+        DataType::Float64 | DataType::Float32 => Ok(DataType::Float64),
+        // Max precision is 38
+        DataType::Decimal(precision, scale) => {
+            // in the spark, the result type is DECIMAL(min(38,precision+10), s)
+            let new_precision = 38.min(*precision + 10);
```

Comment:
As I mentioned above,  I think it will improve usability if we used symbolic constants in the code rather than hard coded numbers such as `38` and `10`

Something like

```rust
/// Maximum precision for a decimal number
let DECIMAL_MAX_PRECISION: usize = 38;
```